pkg/sources/adapter/azureeventhubsource/adapter.go

```diff
@@ -146,28 +149,50 @@ func NewAdapter(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClie
 
 // Start implements adapter.Adapter.
 func (a *adapter) Start(ctx context.Context) error {
+	go health.Start(ctx)
+
 	connCtx, cancel := context.WithTimeout(ctx, connTimeout)
 	runtimeInfo, err := a.ehClient.GetRuntimeInformation(connCtx)
 	cancel()
 	if err != nil {
-		a.logger.Fatal("GetRuntimeInformation failed:", err)
+		return fmt.Errorf("getting Event Hub runtime information: %w", err)
 	}
 	a.runtimeInfo = runtimeInfo
 
+	a.logger.Info("Starting Event Hub message receivers for partitions ", runtimeInfo.PartitionIDs)
+
 	// listen to each partition of the Event Hub
 	for _, partitionID := range runtimeInfo.PartitionIDs {
 		connCtx, cancel := context.WithTimeout(ctx, connTimeout)
 		_, err := a.ehClient.Receive(connCtx, partitionID, a.handleMessage, eventhub.ReceiveWithLatestOffset())
 		cancel()
 		if err != nil {
-			a.logger.Error("failed to start Event Hub listener:", err)
-			continue
+			a.logger.Errorw("An error occurred while starting message receivers. "+
```

Comment:
```suggestion
			a.logger.Errorw("starting message receivers. "+
```

Reply:
That's not a returned error, we're **logging** the error directly here. That message won't be wrapped into another error such as "Error running the adapter: ...", it's the entire message.
Another error is actually _returned_ below.